### PR TITLE
quote paths which might have whitespace

### PR DESCRIPTION
**Pull request by tfoote**
Connects to ros2/ros2#20

### CODE REVIEW

File: rosidl_typesupport_opensplice_cpp/rosidl_typesupport_opensplice_cpp/__init__.py [hunk 1]
```diff
@@ -30,14 +30,15 @@ def generate_dds_opensplice_cpp(
 
         cmd = [idl_pp]
         for include_dir in include_dirs:
-            cmd += ['-I', include_dir]
+            cmd += ['-I', '"%s"' % include_dir]
```
Comment (wjwwood) [inline rosidl_typesupport_opensplice_cpp/rosidl_typesupport_opensplice_cpp/__init__.py]: Just out of curiosity did you try `cmd += ['-I"%s" % include_dir']` or even `cmd += ['"-I%s" % include_dir']`?

**Comment by dirk-thomas** [comment]
Why is this necessary at all? Python subprocess already gets the command in a list where each argument is separated from each other.

File: rosidl_typesupport_opensplice_cpp/rosidl_typesupport_opensplice_cpp/__init__.py (hunk 1 above)
Comment (tfoote) [inline rosidl_typesupport_opensplice_cpp/rosidl_typesupport_opensplice_cpp/__init__.py]: I did not try those. I kept the resultant syntax the same with the space betwen -I and the value.

**Comment by tfoote** [comment]
I don't think this is explicitly necessary for this program. However it seems to pass through the values as passed to the child process a little better. I think the child process is stringifying the arguments and then string processing them. 

Without this you get the following errors; (note CMD IS is a debugging output I added)
```
CMD IS ['/usr/bin/idlpp', '-I', '/home/tfoote/work/r2/ros2 ws/build/simple_msgs/rosidl_generator_dds_idl', '-S', '-l', 'cpp', '-o', 'dds-types', '-d', '/home/tfoote/work/r2/ros2 ws/build/simple_msgs/rosidl_typesupport_opensplice_cpp/simple_msgs/dds_opensplice', '/home/tfoote/work/r2/ros2 ws/build/simple_msgs/rosidl_generator_dds_idl/simple_msgs/dds_opensplice/Uint32_.idl']
Failed to open IDL file: ws/build/simple_msgs/rosidl_generator_dds_idl
Failed to open IDL file: ws/build/simple_msgs/rosidl_generator_dds_idl
simple_msgs /home/tfoote/work/r2/ros2 ws/src/ros2/examples/simple_msgs/msg/Time.msg
```

We could close this as won't fix since it's only a parital solution to the real upstream issue. https://github.com/PrismTech/opensplice/issues/15

File: rosidl_typesupport_opensplice_cpp/rosidl_typesupport_opensplice_cpp/__init__.py (hunk 1 above)
Comment (dirk-thomas) [inline rosidl_typesupport_opensplice_cpp/rosidl_typesupport_opensplice_cpp/__init__.py]: We should try of the tool handles the cases @wjwwood suggested better.

**Comment by tfoote** [comment]
We should probably hold this until we get a fix from upstream. It shouldn't be necessary, and is only a partial fix.

**Comment by dirk-thomas** [comment]
Agreed.

**Comment by esteve** [comment]
+1

**Comment by dirk-thomas** [comment]
Closing in favor of patching upstream: ros2/ros2#35

**Pull request closed without merging**